ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -83,7 +83,7 @@ impl<'a> TraceBuilder<'a> {
         // PHASE 1:
         // Find the control point call and extract the trace inputs struct from its operands.
         let mut inst_iter = blk.instrs.iter().enumerate().rev();
-        for (_, inst) in &mut inst_iter {
+        while let Some((_, inst)) = inst_iter.next() {
```

Comment:
How about:
```
for (_, inst) in inst_iter {
```

(just delete mut?)

Reply:
I must admit that I find `while let` harder to read, but I won't die on this hill.